src/bun.js/api/bun/dns_resolver.zig

```diff
@@ -979,7 +979,10 @@ pub fn CAresLookup(comptime cares_type: type, comptime type_name: []const u8) ty
         next: ?*@This() = null,
         name: []const u8,
 
-        pub usingnamespace bun.New(@This());
+        pub fn new(data: @This()) *@This() {
+            bun.assert(data.allocated); // deinit will not free this otherwise
```

Comment:
We should remove `.allocated` in the future. It exists because we have allocations going through `globalThis.allocator` and `bun.default_allocator.destroy` in `ResolveInfoRequest.init` and `onCaresComplete`